rust/lance/src/dataset/write/update.rs

```diff
@@ -246,17 +246,23 @@ impl UpdateJob {
             WriteParams::with_storage_version(version),
         )
         .await?;
-
         // Apply deletions
         let removed_row_ids = Arc::into_inner(removed_row_ids)
             .unwrap()
             .into_inner()
             .unwrap();
         let (old_fragments, removed_fragment_ids) = self.apply_deletions(&removed_row_ids).await?;
 
+        let num_updated_rows = new_fragments
+            .iter()
+            .map(|f| f.physical_rows.unwrap_or_default() as u64)
```

Comment:
```suggestion
            .map(|f| f.physical_rows.unwrap() as u64)
```
It might be better to just `unwrap` here.  If `physical_rows` is `None` then our assumptions our incorrect and it would be better to return an error than a potentially misleading value.